src/GitHub.VisualStudio/Menus/MenuBase.cs

```diff
@@ -46,6 +46,24 @@ protected MenuBase(IGitHubServiceProvider serviceProvider)
             apiFactory = new Lazy<ISimpleApiClientFactory>(() => ServiceProvider.TryGetService<ISimpleApiClientFactory>());
         }
 
+        protected ILocalRepositoryModel GetRepositoryByPath(string path)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(path))
+                {
+                    var repo = ServiceProvider.TryGetService<IGitService>().GetRepository(path);
+                    return new LocalRepositoryModel(repo.Info.WorkingDirectory.TrimEnd('\\'));
```

Comment:
Interesting that `TrimEnd('\\')` is needed here. Seems to be a failure in the logic in `LocalRepositoryModel.GenerateUrl`. Getting a common subpath in C# isn't easy and seems we're not doing it correctly there. However, for the moment trimming the path here is a good workaround - we should fix that method in another PR.